monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1397 @@
+            legendFormat="Config Server up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["mongos"]),
+            legendFormat="MongoS up:",
+        ),
+        Target(
+            expr=MongoDBMetrics.up(JOBS["shards"]),
+            legendFormat="Shard Server up:",
+        ),
+    ],
+)
+
+version_panel = Stat(
+    title="Version",
+    dataSource=DATASOURCE,
+    format="string",
+    colorMode="none",
+    orientation="horizontal",
+    textMode="auto",
+    reduceOptions={"calcs": ["lastNotNull"], "fields": "/^mongodb$/", "values": False},
+    textSize=20,
+    wideLayout=True,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.version_info(),
+            format="table",
+            instant=True,
+            legendFormat="{{mongodb}}",
+        )
+    ],
+)
+
+disk_usage_pie = PieChart(
+    title="Disk usage",
+    dataSource=DATASOURCE,
+    displayLabels=["value", "name", "percent"],
+    legendDisplayMode="hidden",
+    pieType="pie",
+    unit=UNITS.BYTES,
+    targets=[
+        Target(
+            expr=MongoDBMetrics.db_stats("indexSize"),
+            legendFormat="Indexes",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsUsedSize")} - {MongoDBMetrics.db_stats("indexSize")}',
+            legendFormat="Documents",
+        ),
+        Target(
+            expr=f'{MongoDBMetrics.db_stats("fsTotalSize")} - {MongoDBMetrics.db_stats("fsUsedSize")}',
+            legendFormat="Free",
+        ),
+    ],
+)
+
+avg_doc_size = mongodb_stat(
+    "Average document size",
+    expr=f'avg(mongodb_dbstats_avgObjSize{{database!~"admin|config|local", rs_state="1", job="{JOBS["shards"]}"}})',
+    unit=UNITS.BYTES,
+    decimals=3,
+    description="Average size of documents in the database",
+)
+
+num_docs_per_shard = mongodb_stat(
+    "Number of documents per shard",
+    expr='mongodb_dbstats_objects{rs_state="1", database!~"admin|config|local", rs_nm=~"$rs_nm"}',
+    textMode="value",
+    noValue="0",
+    colorMode="value",
+    thresholds=[Threshold("blue", 0, 0.0)],
+)
+
+size_collections = mongodb_stat(
+    "Size of Collections",
+    unit=UNITS.BYTES,
+    description="MongoDB stores documents in collections. Collections are analogous to tables in relational databases.",
+    targets=[
+        Target(
+            expr=f'max(mongodb_dbstats_storageSize{{database!~"admin|config|local", job="{JOBS["shards"]}"}})',
+            format="table",
+            instant=True,
+            legendFormat="{{collection}}",
+        )
+    ],
+)
+
+index_size = mongodb_stat(
+    "Index size",
+    expr='max by(rs_nm) (mongodb_dbstats_indexSize{rs_state="1", database!~"admin|config|local", rs_nm=~"$rs_nm"})',
+    unit=UNITS.BYTES,
+)
+
+# Activity indicators for overview
+current_operations = mongodb_stat(
+    "Current Operations/sec",
+    expr=f'sum(rate(mongodb_ss_opcounters{{job=~"{JOBS["all_shards"]}", rs_nm=~"$rs_nm"}}[5m]))',
```

Comment:
time range should be `$rate_interval`, to scale appropriately with the currently selected time frame?